src/MarketPlace.sol

```diff
@@ -165,39 +152,22 @@ contract MarketPlace {
         );
 
         {
-            // create the principal tokens array
-            address[9] memory market = [
-                illuminateToken, // Illuminate
-                t[0], // Swivel
-                t[1], // Yield
-                t[2], // Element
-                t[3], // Pendle
-                t[4], // Tempus
-                t[5], // Sense
-                t[6], // APWine
-                t[7] // Notional
-            ];
-
-            // Set the market
-            markets[u][m] = market;
-
-            // Have the lender contract approve the several contracts
-            ILender(lender).approve(u, a, e, t[7], sensePeriphery);
-
-            // Allow converter to spend interest bearing asset
-            if (t[5] != address(0)) {
-                IRedeemer(redeemer).approve(h);
-            }
-
-            // Approve interest bearing token conversion to underlying for APWine
-            if (t[6] != address(0)) {
-                address futureVault = IAPWineToken(t[6]).futureVault();
-                address interestBearingToken = IAPWineFutureVault(futureVault)
-                    .getIBTAddress();
-                IRedeemer(redeemer).approve(interestBearingToken);
+            // assign values for the principal tokens and adapters array
+            for (uint i = 0; i < t.length; i++) {
```

Comment:
I dont immediately see anything wrong here, but will review. The i+1 is because the 0 index is the illuminate PT yea?

Reply:
Yup, that's exactly right